src/zkevm_specs/evm/execution/end_tx.py

```diff
@@ -36,14 +41,17 @@ def end_tx(instruction: Instruction):
 
     # constrain tx status matches with `PostStateOrStatus` of TxReceipt tag in RW
     instruction.constrain_equal(
-        is_persistent, instruction.tx_receipt_write(tx_id, TxReceiptFieldTag.PostStateOrStatus)
+        (1 - is_tx_invalid) * is_persistent,
+        instruction.tx_receipt_write(tx_id, TxReceiptFieldTag.PostStateOrStatus),
     )
 
     # constrain log id matches with `LogLength` of TxReceipt tag in RW
     log_id = instruction.tx_receipt_write(tx_id, TxReceiptFieldTag.LogLength)
     instruction.constrain_equal(log_id, instruction.curr.log_id)
+    # constrain tx status matches with `PostStateOrStatus` of TxReceipt tag in RW
+    if is_tx_invalid:
+        instruction.constrain_zero(log_id)
 
-    # constrain `CumulativeGasUsed` of TxReceipt tag in RW
```

Comment:
Is there a reason this comment was removed?

Reply:
No...add back.